File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        // find rotate pos
        int n = nums.size(), i=0;  
        for(i=1;i<n;i++){
            if(nums[i]<nums[i-1]){
                break;
            }
        }
        if(i==n) return true; 
        for(int j=i+1;(j%n)!=i;j++){
            int curr = j%n; 
            int prev = (j-1)%n;
            if(!(nums[curr]>=nums[prev])) return false;
        }
        return true;
    }
};

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        for i in range(1,len(nums)): 
            if nums[i]<nums[i-1]: 
                break 
        else: 
            return True 
        
        peak_idx = i-1 
        while i%(len(nums))!=peak_idx: 
            if nums[i%len(nums)]>nums[(i+1)%len(nums)]:
                return False 
            i+=1
        return True